packages/amplify-graphql-model-transformer/src/resources/amplify-dynamodb-table/amplify-table-manager-lambda/amplify-table-manager-handler.ts

```diff
@@ -942,6 +967,80 @@ const isKeySchemaModified = (currentSchema: Array<KeySchemaElement>, endSchema:
   return false;
 };
 
+/**
+ * Util function to check if the time to live is modified between old and new table definitions
+ * @param oldTtl TTL config from old table properties
+ * @param endTtl TTL config from input table properties
+ * @returns boolean indicaes the change of TTL
+ */
+export const isTtlModified = (
+  oldTtl: CustomDDB.TimeToLiveSpecificationProperty | undefined,
+  endTtl: CustomDDB.TimeToLiveSpecificationProperty | undefined,
+): boolean => {
+  if (oldTtl === undefined && endTtl === undefined) {
+    return false;
+  }
+  if (oldTtl === undefined || endTtl === undefined) {
+    return true;
+  }
+  return oldTtl.enabled !== endTtl.enabled || oldTtl.attributeName !== endTtl.attributeName;
+};
+
+/**
+ * Get time to live specification for the given table
+ * This API call has a limit rate. The call is staggered randomly within 10s and called with exponential retries
+ * @param tableName table name
+ * @returns time to live specification object
+ */
+const getTtlStatus = async (tableName: string): Promise<DescribeTimeToLiveCommandOutput> => {
+  const initialDelay = Math.floor(Math.random() * 10 * 1000); // between 0 to 10s
+  console.log(`Waiting for ${initialDelay} ms`);
+  await sleep(initialDelay);
+  // Max retry is 3
+  // Retry delay is 1000ms
+  const describeTimeToLiveResult = makeApiCallWithExponentialRetriesOnThrottlingException(
+    async () => await ddbClient.describeTimeToLive({ TableName: tableName }),
+    3,
+    1000,
+  );
+  return describeTimeToLiveResult;
+};
+
+/**
+ * Makes an API call with error retries and exponential backoff.
+ * @param requestData The data or parameters needed for the API call.
+ * @param maxRetries Maximum number of retries before giving up.
+ * @param retryDelayMs Initial delay between retries, in milliseconds.
+ * @returns A promise that resolves with the API call result or rejects after exceeding max retries.
+ */
+const makeApiCallWithExponentialRetriesOnThrottlingException = async <T>(
```

Comment:
Can we reuse our [existing retrier](https://github.com/aws-amplify/amplify-category-api/blob/64a1b7f46ee2edd23aa9364106ad74374ff76aae/packages/amplify-graphql-model-transformer/src/resources/amplify-dynamodb-table/amplify-table-manager-lambda/amplify-table-manager-handler.ts#L969) rather than coding a new one?

Reply:
I've refactored the `retry` method to have exponential backoff

Reply:
Given that the existing retrier doesn't have an initial delay, let's keep the custom retry method.